modules/avatar/react/lib/AvatarButton.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import styled from 'react-emotion';
+import isPropValid from '@emotion/is-prop-valid';
+import Avatar, {AvatarVariant} from './Avatar';
+import {colors} from '@workday/canvas-kit-react-core';
+import {focusRing, hideMouseFocus} from '@workday/canvas-kit-react-common';
+import {SystemIconCircleSize} from '@workday/canvas-kit-react-icon';
+
+export interface AvatarButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  /**
+   * An AvatarVariant enum indicating which variant to use for the default state (Light vs. Dark)
+   */
+  variant: AvatarVariant;
+  /**
+   * An SystemIconCircleSize enum or number value indicating the size of the avatar
+   */
+  size: SystemIconCircleSize | number;
+  /**
+   * Text describing what the avatar is showing
+   */
+  altText: string;
+  /**
+   * The url of the users avatar photo
+   */
+  url?: string;
+  /**
+   * An event handler function that gets called when the avatar is clicked
+   */
+  onClick?: (e: React.SyntheticEvent) => void;
```

Comment:
This is typed incorrectly:
```ts
// `e.currentTarget` is an HTMLButtonElement
<button onClick={e => e.currentTarget.form}></button>

// Typescript error: "Property 'form' does not exist on type 'EventTarget & Element'. ts(2339)"
<AvatarButton onClick={e => e.currentTarget.form}></button>
```

Either we should omit properties already handled by the element we are extending (`React.ButtonHTMLElement<HTMLButtonElement>`) OR type it correctly (so we can control documentation).

```suggestion
  onClick?: React.MouseEventHandler<T>
```

This is how `<button>` is typed internally in React.

Reply:
Thanks, added:
`onClick?: React.MouseEventHandler<HTMLButtonElement>;`